backend/addcorpus/models.py

```diff
@@ -46,6 +60,66 @@ class Meta:
     def __str__(self):
         return self.name
 
+    @admin.display()
+    def ready_to_index(self) -> bool:
+        '''
+        Checks whether the corpus is ready for indexing.
+
+        Runs a try/except around `self.validate_ready_to_index()` and returns a
+        boolean; `True` means the validation completed without errors.
+
+        If you want to see validation error messages, use the validation method
+        directly.
+        '''
+        return _validation_passes(self.validate_ready_to_index)
```

Comment:
Hmm this seems convoluted. Why not just make `_validation_passes` raise a `ValidatonError`?

Reply:
I agree outfactoring `_validation_passes` was a bit unnecessary, if that's what you mean.

The reason why it this method doesn't raise `ValidationError` is that I thought it would be useful to have a method that does not raise errors but just says whether validation passes.

Basically, there are some situations (like trying to index) where you really should raise errors if the corpus is not index-ready, but if you're just displaying the state of the corpus (e.g. in the admin), it's not an "error" if the corpus is incomplete.